File: src/learning_names_cpp/src/name_remap.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node {
public:
    MyNode() : Node("my_node_cpp", "my_namespace_cpp") {
        RCLCPP_INFO(this->get_logger(), "节点创建成功!");
    }
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MyNode>());
    rclcpp::shutdown();

    return 0;
}

File: src/learning_names_py/learning_names_py/name_remap.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class MyNode(Node):
    def __init__(self):
        super().__init__("my_node_py", namespace = "my_namespace_py")
        self.get_logger().info("节点创建成功!")
        # 全局话题
        # self.publisher_ = self.create_publisher(String, "/quanjv", 10)
        # 相对话题
        # self.publisher_ = self.create_publisher(String, "xiangdui", 10)
        # 私有话题
        self.publisher_ = self.create_publisher(String, "~/siyou", 10)

def main():
    rclpy.init()
    node = MyNode()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()

if __name__ == "__main__":
    main()